framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -300,21 +303,28 @@ MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase
   const auto sideset_map = boundary_info.get_sideset_map();
   unsigned int num_faces_without_sideset = 0;
 
+  // Make copy of _watertight_boundaries and sort it for later use
+  std::vector<BoundaryID> boundary_copy = _watertight_boundaries;
+  std::sort(boundary_copy.begin(), boundary_copy.end());
```

Comment:
would not hurt to just store it sorted

Reply:
So is there a reason that the vector of boundary IDs needs to be constant? I was thinking that if it wasn't constant that I could sort it immediately after it's read in and that way it could be stored sorted. Do you think that would cause any issues down the road?

Reply:
I'd leave it non-const if that's the cost to store it sorted without requiring some ugly function in the initializer list.  Everything you can mark `const` is one more clue to future programmers and one less thing they can accidentally break later, so all else being equal I love `const`, but here all else is not all equal.